evennia/contrib/turnbattle/tb_range.py

```diff
@@ -0,0 +1,1356 @@
+
+        Again, this can be expanded upon.
+    """
+    # For this example, just generate a number between 15 and 25.
+    damage_value = randint(15, 25)
+    return damage_value
+
+
+def apply_damage(defender, damage):
+    """
+    Applies damage to a target, reducing their HP by the damage amount to a
+    minimum of 0.
+
+    Args:
+        defender (obj): Character taking damage
+        damage (int): Amount of damage being taken
+    """
+    defender.db.hp -= damage  # Reduce defender's HP by the damage dealt.
+    # If this reduces it to 0 or less, set HP to 0.
+    if defender.db.hp <= 0:
+        defender.db.hp = 0
+
+
+def resolve_attack(attacker, defender, attack_type, attack_value=None, defense_value=None):
+    """
+    Resolves an attack and outputs the result.
+
+    Args:
+        attacker (obj): Character doing the attacking
+        defender (obj): Character being attacked
+        attack_type (str): Type of attack (melee or ranged)
+
+    Notes:
+        Even though the attack and defense values are calculated
+        extremely simply, they are separated out into their own functions
+        so that they are easier to expand upon.
+    """
+    # Get an attack roll from the attacker.
+    if not attack_value:
+        attack_value = get_attack(attacker, defender, attack_type)
+    # Get a defense value from the defender.
+    if not defense_value:
+        defense_value = get_defense(attacker, defender, attack_type)
+    # If the attack value is lower than the defense value, miss. Otherwise, hit.
+    if attack_value < defense_value:
+        attacker.location.msg_contents("%s's %s attack misses %s!" % (attacker, attack_type, defender))
+    else:
+        damage_value = get_damage(attacker, defender)  # Calculate damage value.
+        # Announce damage dealt and apply damage.
+        attacker.location.msg_contents("%s hits %s with a %s attack for %i damage!" % (attacker, defender, attack_type, damage_value))
+        apply_damage(defender, damage_value)
+        # If defender HP is reduced to 0 or less, announce defeat.
+        if defender.db.hp <= 0:
+            attacker.location.msg_contents("%s has been defeated!" % defender)
+            
+def distance_dec(mover, target):
+    """
+    Decreases distance in range field between mover and target.
+    
+    Args:
+        mover (obj): The object moving
+        target (obj): The object to be moved toward
+    """
+    mover.db.Combat_Range[target] -= 1
+    target.db.Combat_Range[mover] = mover.db.Combat_Range[target]
+    # If this brings mover to range 0 (Engaged):
+    if mover.db.Combat_Range[target] <= 0:
+        # Reset range to each other to 0 and copy target's ranges to mover.
+        target.db.Combat_Range[mover] = 0
+        mover.db.Combat_Range = target.db.Combat_Range
+        # Assure everything else has the same distance from the mover and target, now that they're together
+        for object in mover.location.contents:
```

Comment:
`object` is a reserved Python word. Never use a reserved word as a variable name even if you get no error from it here  - not only is it a potential source of insidious bugs, it's also against PEP8 and Python praxis and will also trigger all sorts of warnings in external style checkers down the line. Use something like `obj` instead.

Reply:
Whoops!! I'll definitely fix this.